storage/store.go

```diff
@@ -647,12 +647,14 @@ func (s *Store) MaybeGossipConfigs() error {
 		if _, err := rng.getLeaseForGossip(s.Context(nil)); err != nil {
 			return err
 		}
-		// Always gossip the configs as there are leader lease acquisition
-		// scenarios in which the config is not gossiped (e.g. leader executes
-		// but crashes before gossiping; comes back up but group goes dormant).
-		rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
-			return bytes.Equal(configPrefix, cd.keyPrefix)
-		})
+		// Configs are gossiped immediately when forced. Otherwise, configs
+		// will be gossiped when they are updated or a leader lease is
+		// acquired/extended.
+		if force {
+			rng.maybeGossipConfigs(func(configPrefix proto.Key) bool {
+				return bytes.Equal(configPrefix, cd.keyPrefix)
+			})
+		}
```

Comment:
So, I've changed `NewRange` to call `r.maybeGossipConfigs`. All tests passed.

To summarize, `maybeGossipConfigs` is called from the following three places:

- `NewRange` (= when a range is newly created)
- `InternalLeaderLease` (= when a leader lease is acquired).
- `applyRaftCommand` (= when a config is updated).